doc/examples/timeouts.rst

```diff
@@ -0,0 +1,161 @@
+
+.. _timeout-example:
+
+Client Side Operation Timeout
+=============================
+
+PyMongo 4.2 introduced :meth:`~pymongo.timeout` and the ``timeoutMS``
+URI and keyword argument to :class:`~pymongo.mongo_client.MongoClient`.
+These features allow applications to more easily limit the amount of time that
+one or more operations can execute before control is returned to the app. This
+timeout applies to all of the work done to execute the operation, including
+but not limited to server selection, connection checkout, serialization, and
+server-side execution.
+
+Basic Usage
+-----------
+
+The following example uses :meth:`~pymongo.timeout` to configure a 10-second
+timeout for an :meth:`~pymongo.collection.Collection.insert_one` operation::
+
+  import pymongo
+  with pymongo.timeout(10):
+      coll.insert_one({"name": "Nunu"})
+
+The :meth:`~pymongo.timeout` applies to all pymongo operations within the block.
+The following example ensures that both the `insert` and the `find` complete
+within 10 seconds total, or raise a timeout error::
+
+  with pymongo.timeout(10):
+      coll.insert_one({"name": "Nunu"})
+      coll.find_one({"name": "Nunu"})
+
+When nesting :func:`~pymongo.timeout`, the newly computed deadline is capped to at most
```

Comment:
This was hard to parse, perhaps: "the nested deadline is still capped by the outer deadline".

Reply:
Done.